app/models/concerns/v2/bulk_assignment.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module V2
+  # Logic for bulk reassigning entities to other entity
+  module BulkAssignment
+    extend ActiveSupport::Concern
+
+    included do
+      def bulk_assign(assoc_model, input_list)
+        imported = removed = 0
+
+        transaction do
+          removed = assoc_model.where(model_name.element.to_sym => self).delete_all
+          imported = import_in_bulk!(assoc_model, id, input_list).ids.count if input_list.present?
+        end
+
+        [imported, removed]
+      end
+
+      def import_in_bulk!(assoc_model, left_id, right_ids)
+        right_entities = assoc_model.reflect_on_all_associations.map(&:klass).last.where(id: right_ids)
+
+        raise ActiveRecord::RecordInvalid unless right_entities.size == right_ids.size
```

Comment:
I guess you don't need these if the controller already returns you the entities (see my above comment). But you need some naming changes to args.